addons/service_account_roles.tf

```diff
@@ -0,0 +1,58 @@
+variable "service_account_roles" {
+  type        = "list"
+  default     = []
+  description = "Specify custom IAM roles that can be used by pods on the k8s cluster"
+  # service_account_roles = [
+  #   {
+  #       name  = "foo"
+  #       service_account_namespace = "foo-sa"
+  #       service_account_name = "foo-sa"
+  #       policy = <<-EOF
+  #       IAMPolicyDocument
+  #         WithIdents
+  #       EOF
+  #   }
+  # ]
+}
+
+resource "aws_iam_openid_connect_provider" "example" {
+  count     = (length(var.service_account_roles) > 0) ? 1 : 0
+  client_id_list  = ["sts.amazonaws.com"]
+  thumbprint_list = []
+  url             = "${data.aws_eks_cluster.eks.identity.0.oidc.0.issuer}"
```

Comment:
Where is this coming from? Should it be part of this add on?

Reply:
Yes it creates Identity provider for OpenID Connect associate with cluster

Reply:
it was defined in `main.tf`, so referencing it here
 
data "aws_eks_cluster" "eks" {
  name = var.cluster_name
}